src/components/App.tsx

```diff
@@ -4387,6 +4392,85 @@ class App extends React.Component<AppProps, AppState> {
     return true;
   };
 
+  private handleCanvasZoomUsingCtrlAndSpace = (
+    event: React.PointerEvent<HTMLElement>,
+  ): boolean => {
+    if (
+      !(
+        gesture.pointers.size <= 1 &&
+        (event.button === POINTER_BUTTON.WHEEL ||
+          (event.button === POINTER_BUTTON.MAIN &&
+            isHoldingSpace &&
+            event.ctrlKey) ||
+          isHandToolActive(this.state) ||
+          this.state.viewModeEnabled)
+      ) ||
+      isTextElement(this.state.editingElement)
+    ) {
+      return false;
+    }
```

Comment:
this thing is a little bit hard to read. Perhaps split it into variables, and remove the negation of the whole group?

Also I'm not sure if `ctrl + drag` with hand tool or in view mode should work alone without spacebar (also that condition above doesn't explicitly allow just that, it seems to work by circumstance), for consistency.

Reply:
> [0ec6d9a](https://github.com/excalidraw/excalidraw/pull/6760/commits/0ec6d9ad5d5c58d344a85929f89b9eacd181c1d0)
Yes it was kind of hard to decipher. Fixed it in the last commit.

> Also I'm not sure if ctrl + drag with hand tool or in view mode should work alone without spacebar (also that condition above doesn't explicitly allow just that, it seems to work by circumstance), for consistency.
It seems this issue is also happening with spacebar panning. When you release the spacebar it will still pan until you release the mouse button.
Maybe this needs to be a separate issue since the behaviour is not related to current PR?